core/src/main/java/com/graphhopper/storage/GraphEdgeIdFinder.java

```diff
@@ -200,8 +213,8 @@ public final boolean intersects(EdgeIteratorState edgeState) {
             BBox bbox = null;
             for (int shapeIdx = 0; shapeIdx < blockedShapes.size(); shapeIdx++) {
                 GHIntHashSet blockedEdges = edgesList.get(shapeIdx);
-                // blockedEdges acts as cache that is only useful when filled and for non-virtual edges
-                if (!blockedEdges.isEmpty() && edgeState.getEdge() < baseEdgeCount) {
+                // blockedEdges acts as cache if present and is only useful for non-virtual edges
+                if (blockedEdges != null && edgeState.getEdge() < baseEdgeCount) {
```

Comment:
Throwing an exception for empty Sets sounds good

Reply:
Hm. I just thought about this. The problem then is that users might rely on the silence here. E.g. if users draw arbitrary areas to exclude and before accepting a new blocked area they would need to query us via a route, which sounds ugly. Also it might happen that the area does work now but fails in the future because road is under construction or certain roads are excluded by default and then all routing requests with the empty block area will fail.